packages/zoe/src/zoeService/createZCFVat.js

```diff
@@ -5,14 +5,24 @@ import { E } from '@agoric/eventual-send';
  * ZCF Vats can be created.
  *
  * @param {VatAdminSvc} vatAdminSvc
- * @param {string=} zcfBundleName
+ * @param {ZCFSpec} zcfSpec
  * @returns {CreateZCFVat}
  */
-export const setupCreateZCFVat = (vatAdminSvc, zcfBundleName = 'zcf') => {
+export const setupCreateZCFVat = (vatAdminSvc, zcfSpec) => {
   /** @type {CreateZCFVat} */
   const createZCFVat = async () => {
-    assert.typeof(zcfBundleName, 'string');
-    const rootAndAdminNodeP = E(vatAdminSvc).createVatByName(zcfBundleName);
+    let bundlecapP;
+    if (zcfSpec.bundlecap) {
+      bundlecapP = Promise.resolve(zcfSpec.bundlecap);
```

Comment:
The `await` below already does an implicit `Promise.resolve`.

```suggestion
      bundlecapP = zcfSpec.bundlecap;
```

Reply:
Is it ok (style-wise) for `bundlecapP` to sometimes contain a non-promise? My original version did the `await` inside the two `else` branches and had `bundlecap` (a non-promise) as the shared variable; this was my attempt to obey the "only top-level await" rule, but I assumed there was also a "no inaccurate hungarian variable names" rule :).

Reply:
I applaud your conformance to the `await` rule.

We do have a typing convention for `T or Promise<T>` which is `ERef<T>`. However, we don't have any associated naming convention. I'm sure I've sometimes said `fooP` and sometimes said `foo`. I do not want to introduce a `fooE`.

I don't like the hungarian violation, but I don't like any of the alternatives better. Perhaps also declare the variable to be an `ERef<something>`?